File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        remainders = {}

        for i in range (len(nums)):
            if nums[i] in remainders:
                return [remainders[nums[i]], i]
            remainders[target - nums[i]] = i

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int l = nums.size();
        if (l == 0 || l == 1) {
            return false;}
        std::sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size() - 1; ++i) {
            if (nums[i] == nums[i+1]) {
                return true;}}
       return false; 
    }
};